### DISCUSSION TITLE
Error using gridstat @<time> vs using level for both metplus 5.0.1 and metplus 5.1.0-beta 1 containers

### DISCUSSION THREAD

**Discussion by cpapadop1** [dtcenter/METplus#2172] [Configuration] [answered]
Short version:

For my config file I have the following difference in METplus/MET:

MET:

diff GridStatConfig_default GridStatConfig_default.works
67c67
<         level      = [ "(@20190104,*,*)" ];
---
>         level      = [ "(3,*,*)" ];

METPlus:

diff GridStat_basic.conf  GridStat_basic_fails.conf 
73,74c73
< #FCST_VAR1_LEVELS = "(@20190104,*,*)"
< FCST_VAR1_LEVELS = "(3,*,*)"
---
> FCST_VAR1_LEVELS = "(@20190104,*,*)"

The level version works, the @ version gives me the following error:

GSL_RNG_TYPE=mt19937
GSL_RNG_SEED=18446744073197542756
WARNING:
WARNING: MetNcCFDataFile::convert_time_to_offset() -> 20190104 does not exist at time variable
WARNING:
WARNING:
WARNING: MetNcCFDataFile::data_plane(VarInfo &, DataPlane &) -> the requested time 19700822_162144 for "prcp" variable does not exist (20190101_000000 and 20191231_000000).
WARNING:
WARNING:
WARNING: read_data_plane() -> prcp(@20190104,*,*) not found in file: /METplus-5.0.0/Bug-report_files/cmip5_2019_precip.nc
WARNING:
ERROR  :
ERROR  : process_scores() -> no requested data found!  Exiting...
ERROR  :

NOTE: There is a strange reference to 

the requested time 19700822_162144 for "prcp

Longer version

I then tried the same using just MET with similar results.  Here are the 2 different logs showing the  working and non-working version the only thing I changed was the above mentioned level change.
For record version: level      = [ "(3,*,*)" ];

GSL_RNG_TYPE=mt19937
GSL_RNG_SEED=18446744072098696036
DEBUG 1: Forecast File: Input_files/MET-focusgroup_week3/cmip5_2019_precip.nc
DEBUG 1: Observation File: Input_files/MET-focusgroup_week3/cmip5_2019_precip.nc
DEBUG 4: NcCfFile::getData() -> setting the unset init time to the valid time of 20190104_000000.
*************** Here is where they differ ******************************
DEBUG 4: NcCfFile::getData() -> setting the unset init time to the valid time of 20190104_000000.

For @ version:level      = [ "(@20190104,*,*)" ];

GSL_RNG_TYPE=mt19937
GSL_RNG_SEED=18446744071813548900
DEBUG 1: Forecast File: Input_files/MET-focusgroup_week3/cmip5_2019_precip.nc
DEBUG 1: Observation File: Input_files/MET-focusgroup_week3/cmip5_2019_precip.nc
*************** Here is where they differ ******************************
WARNING:
WARNING: MetNcCFDataFile::convert_time_to_offset() -> 20190104 does not exist at time variable
WARNING:
WARNING:
WARNING: MetNcCFDataFile::data_plane(VarInfo &, DataPlane &) -> the requested time 19700822_162144 for "prcp" variable does not exist (20190101_000000 and 20191231_000000).
WARNING:
WARNING:
WARNING: read_data_plane() -> prcp(@20190104,*,*) not found in file: Input_files/MET-focusgroup_week3/cmip5_2019_precip.nc
WARNING: 

So clearly the issue is somewhere in MetNcDataFile

I tried to attach the pertinent files  I converted the docker containers from the site to singularity containers, but that should not cause an issue.  I am fine using the level, just wanted to report the issue as a possible bug issue.

thanks

Caroline
[Bug-report_files.zip](https://github.com/dtcenter/METplus/files/11457071/Bug-report_files.zip)

**Comment by j-opatz** [comment]
Hi Caroline,

And thank you for your question. 

As you and I have already dug into on this, it seems to be an installation issue or a dependency on something that's activated with the "@" symbol usage for non-directional dimensions and MET just isn't finding what it needs.

I'm going to tag @jprestop here as she's our team specialist for installation concerns. Julie, I've got a copy of all of the files Caroline is using, including her log file set to DEBUG and her configuration file. I tested things out by updating her configuration file input file paths to work with my Seneca paths and with no other changes, it works for me. Nowhere in her configuration file is there a reference to 1979 and the input file itself is CF-compliant, so MET grabs the date correctly for me with MET 11.0, 11.1 betas, etc. 

Let me know what you need from my side to troubleshoot this, and of course feel free to ask Caroline for any settings she has with her Singularity containers.

**Reply by cpapadop1** [reply]
Thanks John, it could be something with the docker/singularity container, I translated the docker container to a singularity container, but, can't try the native docker container since I can't run as root.  I am happy to answer any questions, I also have access to Cheyenne if that helps to debug.  Also I am using the MET that comes with the METplus docker container, not stand alone MET ??

**Reply by jprestop** [reply]
Hi @j-opatz and @cpapadop1.  I'm not convinced this is an installation problem, but it could be.  I was not familiar with the "@" notation, but I found the documentation [here](https://met.readthedocs.io/en/latest/Users_Guide/config_options.html?highlight=netcdf).  Can one of you please provide the commands and config files for the working and not working runs only using MET and not METplus?  I'd like to try running this on Cheyenne.  Thanks!

**Reply by j-opatz** [reply]
Hi Julie, sorry I'm just seeing this now. I'll provide you some Slack context and get you set up with test running this.

**Reply by jprestop** [reply]
Hi @j-opatz and @cpapadop1.  After reproducing the same error that Caroline received on cheyenne, I decided to move the files to seneca and try it there.  I was also able to reproduce the same error she is receiving on seneca using /usr/local/METplus.  However, @j-opatz, you mentioned that you had a successful run on seneca, so I'm not quite sure why that is.  If you'd like to take a look at my files and log output on seneca, you can find them here:
/d1/projects/METplus/discussions/2172/Bug-report_files
/d1/projects/METplus/discussions/2172/Bug-report_files/joutput/logs/metplus.log.20230516184123

**Comment by cpapadop1** [comment]
Hi everyone,

sorry was out yesterday, the setup should be in the zip file that I
attached.  Again this might be specific to the docker container, I can't
run docker, but, did convert to singularity.  Since everything else ran
fine with the container, I am guessing that the conversion to singularity
is not the issue.  John is very familiar with the issue since I was
working with him when it came up.

cheers,

Caroline


On Tue, May 16, 2023 at 9:43 AM j-opatz ***@***.***> wrote:

> Hi Julie, sorry I'm just seeing this now. I'll provide you some Slack
> context and get you set up with test running this.
>
> —
> Reply to this email directly, view it on GitHub
> <https://urldefense.com/v3/__https://github.com/dtcenter/METplus/discussions/2172*discussioncomment-5918757__;Iw!!Mih3wA!CYnrWxhFM_dXoaSpfKhz_tYz4iF6eM1mrGovZ0GM22j3t1TWRpSk5l27kgYZ3CA4yJsSmnGdgPfTtDVJ4ld2YzuXCg$>,
> or unsubscribe
> <https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/ADMHZVAG4HKTKWGRL7HATM3XGOVCPANCNFSM6AAAAAAX62GEDU__;!!Mih3wA!CYnrWxhFM_dXoaSpfKhz_tYz4iF6eM1mrGovZ0GM22j3t1TWRpSk5l27kgYZ3CA4yJsSmnGdgPfTtDVJ4le-gPhYZA$>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Caroline Papadopoulos, PhD
Climate, Atmospheric Science and Physical Oceanography (CASPO)
Center for Western Weather and Water Extremes (CW3E)
Scripps Institution of Oceanography, UCSD
9500 Gilman Drive  Department 0230
La Jolla, CA  92093-0230
858.822.2726
***@***.***

**Comment by jprestop** [comment, marked as answer]
Hi @j-opatz. Thanks for the Slack message pointing me to your /d1/personal/jopatz/workbench/GridStat.conf file on seneca.  I copied your file to my directory and ran with it, but I did not get a successful run and got the same error that Caroline and I were receiving previously.  This led me to check the version of MET that you were using.  I see you were using “/d1/projects/MET/MET_regression/develop/NB20230516/MET-develop” (V11.1.0 - in develop) on seneca, but I was using “/usr/local/met” (V11.0.1).  When I changed the version of MET to use the same version of develop that you were using, I got a successful run.  

It sounds like that this was a bug in MET-11.0.1 that was resolved in a beta version in our develop branch.  Looking at the [MET Version 11.1.0-beta2 release notes](https://met.readthedocs.io/en/develop/Users_Guide/release-notes.html#met-version-11-1-0-beta2-release-notes-20230505), this bugfix jumps out as the possible fix: "[Bugfix: Fix support for the YYYYMMDD format in NetCDF level timestrings (#2482)](https://github.com/dtcenter/MET/issues/2482)."  It looks like it could be related to the problem Caroline is experiencing as I see a [comment](https://github.com/dtcenter/MET/issues/2482#issuecomment-1453878700) in the issue that refers back to [Issue 1815](https://github.com/dtcenter/MET/issues/1815), which led me back to [Refine NetCDF level string handling logic to always interpret @ strings as values #2225](https://github.com/dtcenter/MET/issues/2225).

@cpapadop1, could you please try converting the docker image for the beta2 release to singularity and see if that resolves the problem you are experiencing?  You can access it with "docker pull dtcenter/metplus:5.1.0-beta2" (from this [page](https://hub.docker.com/r/dtcenter/metplus/tags)).

**Comment by cpapadop1** [comment]
Hi everyone,

I have made a singularity container from metplus5.1.0-beta2 and re-run with
the @ level commands and it is now working great !!

FCST_VAR1_LEVELS = ***@***.***,*,*)"
#FCST_VAR1_LEVELS = "(3,*,*)"

So now you can either use the record number or the @<date> format.

this solves the issue I was having,

thanks,

Caroline


On Wed, May 17, 2023 at 9:25 AM jprestop ***@***.***> wrote:

> Hi @j-opatz
> <https://urldefense.com/v3/__https://github.com/j-opatz__;!!Mih3wA!AsMgQAyMaK-hAz9kuZzx4VHDDtxlONZk7WBiqlwI5bcmbHYIEJtQn0x1ZbQFLVLcgDJTAw78ILITUNpit0pBKJE0UA$>.
> Thanks for the Slack message pointing me to your
> /d1/personal/jopatz/workbench/GridStat.conf file on seneca. I copied your
> file to my directory and ran with it, but I did not get a successful run
> and got the same error that Caroline and I were receiving previously. This
> led me to check the version of MET that you were using. I see you were
> using “/d1/projects/MET/MET_regression/develop/NB20230516/MET-develop”
> (V11.1.0 - in develop) on seneca, but I was using “/usr/local/met”
> (V11.0.1). When I changed the version of MET to use the same version of
> develop that you were using, I got a successful run.
>
> It sounds like that this was a bug in MET-11.0.1 that was resolved in a
> beta version in our develop branch. Looking at the MET Version
> 11.1.0-beta2 release notes
> <https://urldefense.com/v3/__https://met.readthedocs.io/en/develop/Users_Guide/release-notes.html*met-version-11-1-0-beta2-release-notes-20230505__;Iw!!Mih3wA!AsMgQAyMaK-hAz9kuZzx4VHDDtxlONZk7WBiqlwI5bcmbHYIEJtQn0x1ZbQFLVLcgDJTAw78ILITUNpit0rRyMBFEQ$>,
> this bugfix jumps out as the possible fix: "Bugfix: Fix support for the
> YYYYMMDD format in NetCDF level timestrings (#2482)
> <https://urldefense.com/v3/__https://github.com/dtcenter/MET/issues/2482__;!!Mih3wA!AsMgQAyMaK-hAz9kuZzx4VHDDtxlONZk7WBiqlwI5bcmbHYIEJtQn0x1ZbQFLVLcgDJTAw78ILITUNpit0ppCA2a6w$>."
> It looks like it could be related to the problem Caroline is experiencing
> as I see a comment
> <https://urldefense.com/v3/__https://github.com/dtcenter/MET/issues/2482*issuecomment-1453878700__;Iw!!Mih3wA!AsMgQAyMaK-hAz9kuZzx4VHDDtxlONZk7WBiqlwI5bcmbHYIEJtQn0x1ZbQFLVLcgDJTAw78ILITUNpit0q4fuXtMw$>
> in the issue that refers back to Issue 1815
> <https://urldefense.com/v3/__https://github.com/dtcenter/MET/issues/1815__;!!Mih3wA!AsMgQAyMaK-hAz9kuZzx4VHDDtxlONZk7WBiqlwI5bcmbHYIEJtQn0x1ZbQFLVLcgDJTAw78ILITUNpit0qlztNQwA$>,
> which led me back to Refine NetCDF level string handling logic to always
> interpret @ strings as values #2225
> <https://urldefense.com/v3/__https://github.com/dtcenter/MET/issues/2225__;!!Mih3wA!AsMgQAyMaK-hAz9kuZzx4VHDDtxlONZk7WBiqlwI5bcmbHYIEJtQn0x1ZbQFLVLcgDJTAw78ILITUNpit0qAxU3w7A$>
> .
>
> @cpapadop1
> <https://urldefense.com/v3/__https://github.com/cpapadop1__;!!Mih3wA!AsMgQAyMaK-hAz9kuZzx4VHDDtxlONZk7WBiqlwI5bcmbHYIEJtQn0x1ZbQFLVLcgDJTAw78ILITUNpit0oi48Aj_A$>,
> could you please try converting the docker image for the beta2 release to
> singularity and see if that resolves the problem you are experiencing? You
> can access it with "docker pull dtcenter/metplus:5.1.0-beta2" (from this
> page
> <https://urldefense.com/v3/__https://hub.docker.com/r/dtcenter/metplus/tags__;!!Mih3wA!AsMgQAyMaK-hAz9kuZzx4VHDDtxlONZk7WBiqlwI5bcmbHYIEJtQn0x1ZbQFLVLcgDJTAw78ILITUNpit0oqmipdPw$>
> ).
>
> —
> Reply to this email directly, view it on GitHub
> <https://urldefense.com/v3/__https://github.com/dtcenter/METplus/discussions/2172*discussioncomment-5929794__;Iw!!Mih3wA!AsMgQAyMaK-hAz9kuZzx4VHDDtxlONZk7WBiqlwI5bcmbHYIEJtQn0x1ZbQFLVLcgDJTAw78ILITUNpit0pvpJBPEg$>,
> or unsubscribe
> <https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/ADMHZVF4H6LIERHLKACRWVTXGT3WXANCNFSM6AAAAAAX62GEDU__;!!Mih3wA!AsMgQAyMaK-hAz9kuZzx4VHDDtxlONZk7WBiqlwI5bcmbHYIEJtQn0x1ZbQFLVLcgDJTAw78ILITUNpit0qngq5tYw$>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Caroline Papadopoulos, PhD
Climate, Atmospheric Science and Physical Oceanography (CASPO)
Center for Western Weather and Water Extremes (CW3E)
Scripps Institution of Oceanography, UCSD
9500 Gilman Drive  Department 0230
La Jolla, CA  92093-0230
858.822.2726
***@***.***

**Reply by jprestop** [reply]
That's great news @cpapadop1.  Thank you for letting us know. I'm going to go ahead and mark this Discussion as answered and lock it.  Please feel free to start a new Discussion with any other questions you have.
